component/src/main/java/com/celements/photo/container/ImageUrl.java

```diff
@@ -0,0 +1,227 @@
+package com.celements.photo.container;
+
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.concurrent.Immutable;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.celements.model.context.ModelContext;
+import com.celements.photo.exception.IllegalImageUrlException;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.Utils;
+
+@Immutable
+public final class ImageUrl {
+
+  // TODO CELDEV-466 There are some cases where the builder is not as predictable and flexible as
+  // one might expect. For details see the improvement issue
+  public static final class Builder {
+
+    private String url;
+    private String action;
+    private String space;
+    private String name;
+    private String filename;
+    private String query;
+    private Integer width;
+    private Integer height;
+
+    public @NotNull Builder url(@NotNull String url) throws IllegalImageUrlException {
```

Comment:
:+1: change/add the two constructors now (because this changes the usage essentialy). Pospone validation on build-method as suggested by @msladek